### PR TITLE
Skip test if a non-dependency module is not installed.

### PR DESCRIPTION
**Pull request by IgorBaratta**

### CODE REVIEW

File: python/test/unit/mesh/test_higher_order_mesh.py [hunk 1]
```diff
@@ -22,6 +19,11 @@
 from ufl import dx
 
 
+scipy = pytest.importorskip("scipy")
+sympy = pytest.importorskip("sympy")
```
Comment (garth-wells) [inline python/test/unit/mesh/test_higher_order_mesh.py]: FIAT depends on SymPy, so I'm surprised that this is needed.

File: python/test/unit/mesh/test_higher_order_mesh.py [hunk 2]
```diff
@@ -22,6 +19,11 @@
 from ufl import dx
 
 
+scipy = pytest.importorskip("scipy")
```
Comment (garth-wells) [inline python/test/unit/mesh/test_higher_order_mesh.py]: I think we could make SciPy a test dependency.

File: python/test/unit/mesh/test_higher_order_mesh.py (hunk 1 above)
Comment (IgorBaratta) [reply python/test/unit/mesh/test_higher_order_mesh.py]: Indeed, it's not really needed for SymPy.

File: python/test/unit/mesh/test_higher_order_mesh.py (hunk 2 above)
Comment (IgorBaratta) [reply python/test/unit/mesh/test_higher_order_mesh.py]: I'm not sure how to make SciPy a test dependency.
Using requirements files, maybe? eg: `test_requirements.txt`

**Comment by IgorBaratta** [comment]
Out-of-date.

**Pull request closed without merging**